vignettes/generic.Rmd

```diff
@@ -18,363 +18,490 @@ library(admiraldev)
-shown within each function page.
+This vignette provides an overview of the most important admiral functions, the
+generic functions. They do not derive a specific variable or parameter in a
+specific dataset but can be used for many different derivations in many
+different datasets. The most important concepts and some examples are presented
+here. For full details and more examples see the documentation of the functions.
 
 ## Required Packages
 
 The examples in this vignette require the following packages.
 
-For example purpose, the SDTM datasets from `{pharmaversesdtm}` are used.
-
 ```{r, warning=FALSE, message=FALSE}
 library(admiral)
-library(pharmaversesdtm)
-library(dplyr, warn.conflicts = FALSE)
-library(stringr)
 library(tibble)
-
-data("dm")
-data("ds")
-data("ex")
-data("ae")
-dm <- convert_blanks_to_na(dm)
-ds <- convert_blanks_to_na(ds)
-ex <- convert_blanks_to_na(ex)
-ae <- convert_blanks_to_na(ae)
-```
-```{r echo=FALSE}
-# Filter test patients and make more realistic and interesting for the examples
-dm <- filter(dm, USUBJID %in% c("01-701-1111", "01-701-1047", "01-701-1057"))
-ds <- filter(ds, USUBJID %in% c("01-701-1111", "01-701-1047", "01-701-1057")) %>%
-  mutate(DSSTDTC = case_when(
-    USUBJID == "01-701-1111" & DSDECOD == "RANDOMIZED" ~ "2012-08-01",
-    TRUE ~ DSSTDTC
-  ))
-ex <- filter(ex, USUBJID %in% c("01-701-1111", "01-701-1047", "01-701-1057"))
-ae <- filter(ae, USUBJID %in% c("01-701-1111", "01-701-1047")) %>%
-  mutate(AESTDTC = case_when(
-    USUBJID == "01-701-1111" & AESTDY == "-61" ~ "2012-09-14",
-    TRUE ~ AESTDTC
-  )) %>%
-  mutate(AESTDY = case_when(
-    USUBJID == "01-701-1111" & AESTDY == "-61" ~ 8,
-    TRUE ~ AESTDY
-  ))
 ```
 
-# Difference between _merged_ and _joined_ functions
-
-Firstly, we need to address the question all new users will have... why do we have
-"merged" and "joined" functions and are they not the same thing??
-
-Essentially in both sets of functions we are trying to take information from
-an additional dataframe (passed into the `dataset_add` argument) and join it to
-our original dataframe (passed into the `dataset` argument). The subtle difference
-is that "merged" functions select the relevant information from the additional
-dataframe and then adds it to the original dataframe; whilst "joined" functions
-allow an extra filter condition after the join to further select the relevant
-information.
-
-So in `derive_vars_joined()` for example, the filter conditions can depend on variables
-from both dataframes; whilst in `derive_vars_merged()` only variables from the
-additional dataframe can be used. Examples will be shown in the below sections to
-further explain the difference.
-
-In principle though, we actually could achieve every "merged" function result using
-the equivalent "joined" function. However, the "joined" functions require much more
-resources (time and memory), hence why we include both options for users.
```

Comment:
to clarify why we have "merged" function if "joined" function can accomplish the same functionality - should we add this part back?

Reply:
Good point! I have readded it.